src/app/static/src/app/components/forms/CustomForm.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+    <b-form :ref="id" class="custom-form">
+        <div v-for="section in form.controlSections">
+            <h3>{{section.label}}</h3>
+            <p class="text-muted">{{section.description}}</p>
+            <control-group v-for="group in section.controlGroups" :group="group"></control-group>
+        </div>
+        <button class="btn btn-red" v-on:click="submit">Submit</button>
+    </b-form>
+</template>
+
+<script lang="ts">
+
+    import Vue from "vue";
+    import {BForm} from "bootstrap-vue";
+    import {formMeta} from "./fakeFormMeta";
+    import ControlGroup from "./ControlGroup.vue";
+    import {Dictionary} from "vuex";
+
+    export default Vue.extend({
+        name: "CustomForm",
+        props: ["id"],
+        data() {
+            return {
+                form: formMeta
+            }
+        },
+        components: {
+            BForm,
+            ControlGroup
+        },
+        methods: {
+            submit(e: Event) {
+                e.preventDefault();
+                const form = this.$refs[this.id] as HTMLFormElement;
+                const formData = new FormData(form);
+                const data: Dictionary<any> = {};
+                formData.forEach(function (value, key) {
+                    data[key] = value;
+                });
+                const json = JSON.stringify(data);
+                this.$emit("submit", json);
```

Comment:
The idea is that this component can be pulled out into a standalone js package at some point. The parent component that instantiates it (in our case the ModelRun component) can do what it wants with the json string that's returned. Not sure how this will work if we want to trigger the submit on "continue" though :thinking:

Reply:
In fact maybe this should emit JSON rather than a string, more flexible for the caller?

Reply:
Ok in fact this is easy enough: https://github.com/mrc-ide/hint/pull/98/commits/443fd1fbbed0056f4680b8684637b5fbc28d1ea9